addons/payment_razorpay/models/payment_token.py

```diff
@@ -0,0 +1,53 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, models
+
+from odoo.addons.payment_razorpay import const
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    def _razorpay_get_limit_exceed_warning(self, amount, currency_id):
+        """ Return a warning message when the maximum payment amount is exceeded.
+
+        :param float amount: The amount to be paid.
+        :param currency_id: Currency of the amount.
+        :return: A warning message when the maximum payment amount is exceeded.
+        :rtype: str
+        """
+        self.ensure_one()
+
+        if not amount or self.provider_code != 'razorpay':
+            return ""
+
+        Transaction = self.env['payment.transaction']
+        # Find first transaction from which this token was created.
+        primary_token_tx = Transaction.search(
+            [('token_id', '=', self.id), ('operation', 'not in', ['offline', 'online_token'])],
+            limit=1,
+        )
+        if primary_token_tx:
+            mandate_max_amount = primary_token_tx._razorpay_get_mandate_max_amount()
+        else:
+            # Retrieve the payment method code for this token and get maximum amount based on that.
+            random_token_tx_pm = Transaction.search([
+                ('token_id', '=', self.id),
+            ], limit=1).payment_method_id
```

Comment:
While pushing a commit to improve some comments, I remembered that the payment method is saved on the token too, so I deleted this search.